### PR TITLE
Milestone-3 Feature/friendships v2

### PR DESCRIPTION
**Pull request by Div685**
## Milestone-3 Friendships v2
In this PR we implemented friendship with 2 rows for mutual friendship.
- updated accept method in users_controller to stop reverse pair.
- updated application_controller to check send_request and pending_request for users.

**We also run unit and integrations tests.**
- kindly request you use the order flag while running the integrations tests ` rspec --order defined --format documentation ` 

kindly review the project

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -35,4 +37,16 @@
...
+
+<section class="section-invitation">
+  <% if @user == current_user %>
+    <h2>My Friends</h2>
+    <% @friends_lists.each do |frnd| %>
+      <div class="invitation">
+        <p class="invitation-name"> <span> <%= frnd.user.name %> </span> is your friend</p>
+      </div>
+    <% end %>
+  <% end %>
```
Comment (lekegitrepo) [inline app/views/users/show.html.erb]: - **_Remove all the logic from the view, use view helper instead._**

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -24,6 +24,8 @@
   </article>
 </section>
 
+<div class="invite-friends">
+
 <section class="section-invitation">
   <% if @user == current_user %>
```
Comment (lekegitrepo) [inline app/views/users/show.html.erb]: - **_Remove all the logic from the view, use view helper instead._**

File: app/views/users/_user.html.erb [hunk 3]
```diff
@@ -7,13 +7,16 @@
...
           <%= link_to 'Accept', user_accept_path(user), method: :post, class: "accept-btn" %>
           <%= link_to 'Reject', user_reject_path(user), method: :post, class: "reject-btn" %>
         <% end %>
+
+      <% else %>
+        <span>Friend</span>
+        
       <% end %>
     <% else %>
       <%= link_to "Send Request", user_invite_path(user), method: :post,  class: "send-btn" %>
```
Comment (lekegitrepo) [inline app/views/users/_user.html.erb]: - **_Remove all the logic from the view, use view helper instead._**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! ♻️
### Hi @Div685 and @Adewunmi97 👏 👏
#### You have done a good job so far. The model and controller of the project are well implemented. However, there are some requirements missing/not implemented correctly. Check my comments below 👇

#### 1) LINT:
 - Enable linter for this Pull Request

#### 2) VIEW & LOGIC:
- There should be NO application logic in the views (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.
  * View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
  * There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
    * [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
    * [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
    * [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

#### 3) APP BEHAVIOUR & POSTSCONTROLLER:
 - PostsController#index displays posts in a timeline (their own posts + posts from their friends) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/timeline.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/final-project)). Currently, the app is showing all the posts regardless of whether the post's author is a friend of the current user or not. Fix this the current user should only see the posts of their friends and vice-versa.

#### _Reviewed by Temitayo Adeleke_

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [x] everything works fine, you only need to enable linters :)

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

**Pull request merged**